src/aws_s3_presigned_url.erl

```diff
@@ -0,0 +1,150 @@
+%%%% @doc
+%%%
+%%% Allows generating either a get or put presigned s3 url.
+%%% This can be used by external clients such as cURL to access the object in question.
+%%%
+%%% See:
+%%%   - https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html
+%%%   - https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html
+-module(aws_s3_presigned_url).
+
+-export([ make_presigned_v4_url/5
+        ]).
+
+-include_lib("hackney/include/hackney_lib.hrl").
+
+%%====================================================================
+%% API
+%%====================================================================
+-spec make_presigned_v4_url(map(), get | put, integer(), binary(), binary()) -> {ok, binary()}.
+make_presigned_v4_url(Client0, Method, ExpireSeconds, Bucket, Key) ->
+    MethodBin = aws_request:method_to_binary(Method),
+    Path = ["/", aws_util:encode_uri(Bucket), "/", aws_util:encode_multi_segment_uri(Key), ""],
+    Client = Client0#{service => <<"s3">>},
+    SecurityToken = maps:get(token, Client, undefined),
+    AccessKeyID = maps:get(access_key_id, Client),
+    Region = maps:get(region, Client),
```

Comment:
I think we should create a module the encapsulates the client.

Reply:
As in create utility functions for these acccessors? There already is `aws_client` which would be a good home for this I guess. Unfortunately some of this access is done from `aws-codegenerator` generated code so if it's ok by you I'd prefer cleaning that up separately. 

If you agree, I'll create an issue and follow up on it :ok_hand: